website_react/src/components/dataCollectionForms/dbConnectionForm.jsx

```diff
@@ -0,0 +1,147 @@
+import InputList from '../inputs/InputList'
+
+
+const DBConnectionForm = () => {
+    const validateFunc = () => {
+        console.log('validate');
+    };
+    
+    const data = {
+        fields: [
+            {
+                title: "What is the name of the DB connection?",
+                type: "string",
+                default_value: null,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Select DB connection",
+                type: "enum",
+                default_value: "SQLite",
+                mandatory: true,
+                validatorCallback: validateFunc,
+                options: [
+                    "SQLite",
+                    "Postgresql",
+                    "MySQL",
+                    "Oracle",
+                    "MSSQL",
+                    "Firebird",
+                    "Sybase",
+                    "MongoDB",
+                ],
+            },
+            {
+                title: "Do you want to log queries?",
+                type: "boolean",
+                default_value: false,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Do you want create Database?",
+                type: "boolean",
+                default_value: true,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Enter the pool recycle time",
+                type: "integer",
+                default_value: 3200,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Do you want to set pool pre ping?",
+                type: "boolean",
+                default_value: false,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Enter the port no. of your DB",
+                type: "integer",
+                default_value: null,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Enter host of your DB",
+                type: "string",
+                default_value: "localhost",
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Enter your DB username",
+                type: "string",
+                default_value: "user",
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+            {
+                title: "Enter your DB password",
+                type: "string",
+                default_value: null,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+        ],
+    };
+	return (
+		<>
```

Comment:
code duplication. 
you could to here `return (<Form fields={fields}/>)`.

but just create the JSON lists upfront. and we will pass them each time to a Form component

Reply:
if we have 
1. list of fields 
2. Form component that can render any form that we like 
do we need to add more component to render a new form?